crypto/engine/eng_dyn.c

```diff
@@ -349,28 +349,34 @@ static int dynamic_ctrl(ENGINE *e, int cmd, long i, void *p, void (*f) (void))
     }
     switch (cmd) {
     case DYNAMIC_CMD_SO_PATH:
-        /* a NULL 'p' or a string of zero-length is the same thing */
-        if (p && (strlen((const char *)p) < 1))
-            p = NULL;
         OPENSSL_free(ctx->DYNAMIC_LIBNAME);
-        if (p)
-            ctx->DYNAMIC_LIBNAME = OPENSSL_strdup(p);
-        else
+        /* a NULL 'p' or a string of zero-length is the same thing */
+        if ((p == NULL) || (strlen((const char *)p) < 1)) {
```

Comment:
@FdaSilvaYY Yes, they formally are useless, however there're a lot of code in the project where more parenthesis are preferred to fewer parenthesis.